network/records/handshake_signature.go

```diff
@@ -0,0 +1,27 @@
+package records
+
+import (
+	"crypto/sha256"
+	"strings"
+	"time"
+
+	"github.com/libp2p/go-libp2p/core/peer"
+)
+
+type HandshakeData struct {
+	SenderPeerID    peer.ID
+	RecipientPeerID peer.ID
+	Timestamp       time.Time // Unix timestamp ROUNDED TO 30 SECONDS
+	SenderPubKey    string
```

Comment:
if this is encrypted (which is not a must but you did it in your PR) then we won't be able to access the pubkey to unseal.